CON-381: Overhaul README.md (#124)
rticommunity/rticonnextdds-connector: README.md

# RTI Connector for Connext DDS - Getting Started

*RTI Connext* is a software connectivity framework for real-time distributed applications.
It uses the DDS publish-subscribe communications model to make data distribution efficient and robust.
At its core is the world’s leading ultra-high performance, distributed networking databus.

> *RTI Connector* is a simplified API that wraps RTI Connext Professional's XML Application Creation

Reviewer: ```suggestion
*RTI Connector* is a simplified API that wraps RTI Connext Professional's XML-Based Application Creation
```

> *RTI Connector* is a simplified API that wraps RTI Connext Professional's XML-Based Application Creation

Reviewer: ```suggestion
*RTI Connector* is a simplified API that wraps the *RTI Connext Professional* XML-Based Application Creation
```

> feature where an XML file defines a DDS system.

Reviewer: ```suggestion
feature, which supports the use of XML for the complete system definition.
```

For each participant in the system, you can create a Connector and use its Inputs and Outputs

> (`DataReaders` and `DataWriters`, respectively) to send and receive samples using DynamicData.

Reviewer: ```suggestion
(*DataReaders* and *DataWriters*, respectively) to send and receive samples using DynamicData.
```

![connector-overview.svg](resources/connector-overview.svg)

*RTI Connector* is provided as a library in binary form for selected platforms,
with language bindings and examples provided in source format.

## Language Support

* **[JavaScript](https://github.com/rticommunity/rticonnextdds-connector-js)** (Fully supported)
* **[Go](https://github.com/rticommunity/rticonnextdds-connector-go)** (Experimental support only)
* **[Rust](https://github.com/rticommunity/rticonnextdds-connector-rust)** (Experimental support only)

> Connector for Python has been deprecated. A new full [Connext Python API][python-api] is available.
> The *Connext* [Migration Guide][python-migration] provides guidance to migrate from Connector for Python to the Connext Python API.

[python-api]: https://community.rti.com/static/documentation/connext-dds/current/doc/api/connext_dds/api_python/index.html "Python API documentation"
[python-migration]: https://community.rti.com/static/documentation/connext-dds/current/doc/manuals/migration_guide/700/product700/core700.html#migrating-from-connector-to-the-connext-python-api "Migration guide for Connector for Python to Connext Python API"

## Platform Support

> Platform Support might differ between releases, check the [RTI Connector releases][connector-releases].

Reviewer: ```suggestion
Platform Support may differ between releases. See [RTI Connector releases][connector-releases] for platform availability.
```

[connector-releases]: https://www.github.com/rticommunity/rticonnextdds-connector/releases "RTI Connector releases"

## Support

If you have questions, use the [RTI Community Forum][community-forum] or Github Issues.

> If you have an active support contract with RTI, you can also use the[RTI Support Portal][support-portal].

Author: ```suggestion
If you have an active support contract with RTI, you can also use the [RTI Support Portal][support-portal].
```

> If you have an active support contract with RTI, you can also use the[RTI Support Portal][support-portal].

Reviewer: ```suggestion
If you have an active support contract with RTI, you can also use the [RTI Support Portal][support-portal].
```

[community-forum]: https://community.rti.com/ "RTI Community Forum"
[support-portal]: https://support.rti.com/ "RTI Support Portal"

## License

> *Note:* Each language binding may have its own license.

> > Please refer to the specific language binding repository for more details.

Reviewer: ```suggestion
> Refer to the specific language binding repository for more details.
```

> This software is an experimental ("pre-production") product.

Reviewer: Is RTI Connector experimental when JavaScript is supported? Should this text specify that Connector for Go and Connector for Rust are experimental?

Also, the [Go readme](https://github.com/rticommunity/rticonnextdds-connector-go#rti-connector-for-connext-dds---go) doesn't mention that it's experimental; should the same [note in the Rust readme](https://github.com/rticommunity/rticonnextdds-connector-rust#rti-connector-for-rust-experimental) be added to the Go readme?

Author: > Is RTI Connector experimental when JavaScript is supported? Should this text specify that Connector for Go and Connector for Rust are experimental?

No, as far as I understand it's not. I think that this is the standard text for experimental/non-supported products, whereas the Connector for JavaScript is covered by the "Each language binding may have its own license" note.

> Also, the Go readme doesn't mention that it's experimental; should the same note in the Rust readme be added to the Go readme?

The Connector for Go is maintained by @kyoungho, what are your thoughts on this?

Reviewer 2: Agree. I will add the same note to the Go Connector README.
